api/provider/aws/load_balancer_delete.go

```diff
@@ -44,6 +75,33 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
+	// Check for eventually consistency
+	fnReadSG := func() (shouldRetry bool, err error) {
```

Comment:
we may want to put this into a helper function since environment delete will want to check that its security group got deleted as well.